src/main/java/io/confluent/connect/elasticsearch/ElasticsearchClient.java

```diff
@@ -15,102 +15,568 @@
+        log.debug("Using AK 2.2-2.5 SslFactory methods.");
+
+      } catch (Exception ex) {
+        // must be running AK 2.6+
+        log.debug(
+            "Could not find Ak 2.3-2.5 methods for SslFactory."
+                + " Trying AK 2.6+ methods for SslFactory."
+        );
+        try {
+          sslEngine =
+              SslFactory.class.getDeclaredMethod("sslEngineFactory").invoke(kafkaSslFactory);
+          log.debug("Using AK 2.6+ SslFactory methods.");
+        } catch (Exception exc) {
+          throw new ConnectException("Failed to find methods for SslFactory.", exc);
+        }
+      }
+
+      try {
+        sslContext =
+            (SSLContext) sslEngine.getClass().getDeclaredMethod("sslContext").invoke(sslEngine);
+      } catch (Exception ex) {
+        throw new ConnectException("Could not create SSLContext.", ex);
+      }
+    }
+
+    HostnameVerifier hostnameVerifier = config.shouldDisableHostnameVerification()
+        ? (hostname, session) -> true
+        : SSLConnectionSocketFactory.getDefaultHostnameVerifier();
+
+    builder.setSSLContext(sslContext);
+    builder.setSSLHostnameVerifier(hostnameVerifier);
+    builder.setSSLStrategy(new SSLIOSessionStrategy(sslContext, hostnameVerifier));
+  }
 
   /**
-   * Creates a bulk request for the list of {@link IndexableRecord} records.
+   * Customizes the client according to the configurations.
    *
-   * @param batch the list of records
-   * @return the bulk request
+   * @param builder the HttpAsyncClientBuilder
+   * @return the builder
    */
-  BulkRequest createBulkRequest(List<IndexableRecord> batch);
+  private HttpAsyncClientBuilder customizeHttpClientConfig(HttpAsyncClientBuilder builder) {
+    builder.setMaxConnPerRoute(config.maxInFlightRequests());
+    configureAuthentication(builder);
+
+    if (config.secured()) {
+      log.info("Using secured connection to {}.", config.connectionUrls());
+      configureSslContext(builder);
+    } else {
+      log.info("Using unsecured connection to {}.", config.connectionUrls());
+    }
+    return builder;
+  }
 
   /**
-   * Executes a bulk action.
+   * Customizes each request according to configurations.
    *
-   * @param bulk the bulk request
-   * @return the bulk response
-   * @throws IOException if the client cannot execute the request
+   * @param builder the RequestConfigBuilder
+   * @return the builder
    */
-  BulkResponse executeBulk(BulkRequest bulk) throws IOException;
+  private RequestConfig.Builder customizeRequestConfig(RequestConfig.Builder builder) {
+    return builder
+        .setContentCompressionEnabled(config.compression())
+        .setConnectTimeout(config.connectionTimeoutMs())
+        .setConnectionRequestTimeout(config.readTimeoutMs());
+  }
+
+  /**
+   * Gets the mapping for an index.
+   *
+   * @param index the index to fetch the mapping for
+   * @return the MappingMetaData for the index
+   */
+  private MappingMetaData getMapping(String index) {
```

Comment:
we usually don't use `get/set` in connector-land when naming methods. 
nit: `mapping` since we are doing a general refactoring here and keeping the old name might not worth it.